### DISCUSSION TITLE
Shut down gui without stopping farm

### DISCUSSION THREAD

**Discussion by lavarok** [Chia-Network/chia-blockchain#7264] [Ideas]
It would be great to be able to launch and shutdown the gui without impacting the farm. I dont want to run the gui all the time, so I start everything on the command line. However, currently from the command line there is no way to differentiate what plots are portable vs legacy non nft. For this, one needs to use the gui and review the plots there. Its frustrating that shutting down the gui, shuts down the node/farm/etc.

**Comment by loppefaaret** [comment]
when the GUI exits, it sends a signal to all chia services to close down, nothing to do about that
you can do a `chia plots check -n 5` and see what "pool public key" a plot is tied to - any NFT based plot would be "none"

**Reply by lavarok** [reply]
Surely code can be modified to not send the daemon exit signals and still close the gui. In fact, I might have to submit a merge request for this option. For now file->  "chia-blockchain-giu/src/electron/main.tsx" can be hacked to remove this behavior.

**Reply by loppefaaret** [reply]
yes, but 99% of the GUI users want the current behavior - you are absolutely more than welcome to submit a PR, but just do us a favor and sign it when you do please :)

**Reply by billtorcaso** [reply]
This user wants the GUI and the Daemons to have loose coupling, both as a robust design criterion and as a convenience for me.

Can you supply data to confirm your 99% claim?

**Reply by lavarok** [reply]
Here is an edit that will prevent the gui from shutting down the daemons. Just edit the file diffed below and then do "npm run build" in the chia-blockchain-gui folder. After that, the gui will no longer shutdown daemons on exit. I'll refine this into a PR with the option at exit to choose at a later date.  For now this works. 

```
diff --git a/src/electron/main.tsx b/src/electron/main.tsx
index 7b3d092a..104dedfd 100644
--- a/src/electron/main.tsx
+++ b/src/electron/main.tsx
@@ -205,13 +205,13 @@ if (!handleSquirrelEvent()) {
           }
           isClosing = false;
           decidedToClose = true;
-          mainWindow.webContents.send('exit-daemon');
-          mainWindow.setBounds({ height: 500, width: 500 });
-          ipcMain.on('daemon-exited', (event, args) => {
+          //mainWindow.webContents.send('exit-daemon');
+          //mainWindow.setBounds({ height: 500, width: 500 });
+          //ipcMain.on('daemon-exited', (event, args) => {
             mainWindow.close();
 
             openedWindows.forEach((win) => win.close());
-          });
+          //});
         }
       });
     };
```

**Reply by loppefaaret** [reply]
> This user wants the GUI and the Daemons to have loose coupling, both as a robust design criterion and as a convenience for me.
> 
> Can you supply data to confirm your 99% claim?

millions of installs, a handful of request for this - thats good enough ?

**Comment by lavarok** [comment]
Here is an edit that will prevent the gui from shutting down the daemons. Just edit the file diffed below and then do "npm run build" in the chia-blockchain-gui folder. After that, the gui will no longer shutdown daemons on exit. I'll refine this into a PR with the option at exit to choose at a later date.  For now this works. 

```
diff --git a/src/electron/main.tsx b/src/electron/main.tsx
index 7b3d092a..104dedfd 100644
--- a/src/electron/main.tsx
+++ b/src/electron/main.tsx
@@ -205,13 +205,13 @@ if (!handleSquirrelEvent()) {
           }
           isClosing = false;
           decidedToClose = true;
-          mainWindow.webContents.send('exit-daemon');
-          mainWindow.setBounds({ height: 500, width: 500 });
-          ipcMain.on('daemon-exited', (event, args) => {
+          //mainWindow.webContents.send('exit-daemon');
+          //mainWindow.setBounds({ height: 500, width: 500 });
+          //ipcMain.on('daemon-exited', (event, args) => {
             mainWindow.close();
 
             openedWindows.forEach((win) => win.close());
-          });
+          //});
         }
       });
     };
```

**Reply by kienner-philippe** [reply]
Thanks, good point.
As the GUI is an electron app started with "npm run electron", maybe an even easier hack would be to add
` && chia start farmer`
to the line "electron": in package.json in the script section,
this would restart the services right away.
so instead of 
`"electron": "electron .",`
you would have
`"electron": "electron . && chia start farmer",`
of course you can change the chia command to whatever suits you.
A+
Philippe

**Comment by billtorcaso** [comment]
Sorry, no.

You’ve made an assumption, that requests are uniformly distributed among
all users. If that is not true, then your conclusion is unsupported.

Another perspective is that people only/mostly  complain about things that
are really important.

  — Bill

**Reply by loppefaaret** [reply]
it goes both ways, no ?

**Comment by billtorcaso** [comment]
Yes, it does. I meant to highlight the unstated assumption. Let’s remain
friends and move on.

**Comment by PaJaSoft** [comment]
Hi guys,
I'm probably the second man over the Earth who is missing this behavior like lavarok.

I understand the vast majority of users expects a different behavior so why to bother them right? So what about introducing a new parameter like optional GUI cmdline param (might come to the GUI Settings too) which can introduce this change? That means who doesn't care isn't affected however who wants that (currently missing) behavior appreciate it for sure.

Is there any objection why we can't do that by this way without forking or doing that by other non-cooperative friendly way? 

I don't know chia impl. internals, have no idea what .tsx suffix is at all:-) (what language when Python interpreter is involved?) but at least it looks as a quite easy implementation for  me and I'm willing to help to bring it on the light.

**Comment by ForkFarmer** [comment]
I agree with the suggestion, I don't need/use the GUI most of the time but I would use it more often if I knew I could close it without killing the farm.

**Comment by loppefaaret** [comment]
i can tease you all in this thread a bit, and say that i have looked over some engineers shoulder, and see a function of not killing the processes when the GUI is closing
Whether the function is ever included, or when, i cannot say

**Reply by dfifield1** [reply]
Are they going to implement this? I like the UI, but I am running multiple harvesters, GPUs etc.  It is great to look at the UI to see the status, but it sucks that I can not just shutdown the UI and have the services continue to run.  I am also running the GUI with a remote X server.  This does not seem like it is a very big ask.

**Reply by BrandtH22** [reply]
It might be worthwhile to look into using the chia-exporter as that will likely have much of the information you are using the GUI for without the need to worry about which instance is running what:
https://github.com/Chia-Network/chia-exporter

Video tutorial for setting this up:
https://www.youtube.com/watch?v=7NNbmuwf99s

**Reply by lavarok** [reply]
This is not a replacement for the gui which should be able to connect to background processes, do what you need, and leave them (background processes for node/farm) alone when you close it!

**Comment by BrandtH22** [comment]
Hey @lavarok , exciting news, the upcoming 2.3 release will have the ability to keep services running in the background when shutting down the GUI 🥳

**Reply by lavarok** [reply]
Great news!

**Reply by BrandtH22** [reply]
2.3 was just released with this running services in the background feature:

You can download 2.3 from here: https://www.chia.net/downloads/
This overviews how to use the background services feature: https://docs.chia.net/using-the-gui/#run-chia-services-in-the-background

Let us know if any questions arise and since this feature is now available I will be closing this feature request 👍
